examples/llava/clip.cpp

```diff
@@ -2967,6 +3279,65 @@ bool clip_image_batch_encode(clip_ctx * ctx, const int n_threads, const clip_ima
         }
     }
 
+    if (use_window_attn && ctx->proj_type == PROJECTOR_TYPE_QWEN25VL) {
+        struct ggml_tensor * window_idx = ggml_graph_get_tensor(gf, "window_idx");
+        struct ggml_tensor * inv_window_idx = ggml_graph_get_tensor(gf, "inv_window_idx");
+        struct ggml_tensor * window_mask = ggml_graph_get_tensor(gf, "window_mask");
+
+        const int merge_ratio = 2;
+        const int attn_window_size = 112;
+        const int pw = image_size_width / patch_size / merge_ratio;
+        const int ph = image_size_height / patch_size / merge_ratio;
+        const int grid_window = attn_window_size / patch_size / merge_ratio;
+        const int ipw = image_size_width / patch_size;
```

Comment:
Ok now I realized that this code block looks identical to the one above (which also sets `window_idx`), and this is the problem why 3D model doesn't work.

Still, I have no idea why this code block is duplicated. I'll try to refactor it, but just for sure @HimariO can you confirm if this is NOT intentional?

Reply:
This is not intentional, I think the [duplicated block](https://github.com/HimariO/llama.cpp.qwen2.5vl/blob/qwen25-vl/examples/llava/clip.cpp#L3151-L3201) is leftover when I try to separating the window-attention related code block from Qwen2VL.

Reply:
After look into this futher, the [duplicated block](https://github.com/HimariO/llama.cpp.qwen2.5vl/blob/53a15d014f2dd0a1409e49da097dba891c629f6e/examples/llava/clip.cpp#L3166-L3201) is acutlly used for create `idx` for [rearranging](https://github.com/HimariO/llama.cpp.qwen2.5vl/blob/53a15d014f2dd0a1409e49da097dba891c629f6e/examples/llava/clip.cpp#L3216) `position_ids`. Only the [ggml_graph_get_tensor](https://github.com/HimariO/llama.cpp.qwen2.5vl/blob/53a15d014f2dd0a1409e49da097dba891c629f6e/examples/llava/clip.cpp#L3156) and  `ggml_backend_tensor_set` part is redundant.

Reply:
[Here](https://github.com/HimariO/llama.cpp.qwen2.5vl/blob/cc6a49e478e6e6e2b9901b804f78c01169bcdf63/examples/llava/clip.cpp#L3142-L3265) is an updated version that properly separates qwen2vl and qwen2.5vl logic.

Reply:
I mean, the whole outer block is duplicated in 2 places. I suppose you did this way because you want to re-use the same code for both qwen2 and qwen2.5

In anyway, I ended up separate the code block for qwen2 and qwen2.5 so it looks more clean. For the code of qwen2, I took the version before this PR, so qwen2 should be correct.

But just to be sure, can you take a look at this file to see if the logic for qwen2.5 still look correct? https://github.com/ggml-org/llama.cpp/blob/a264a2546ebd4603f4ef6fe9188ffcb3eb59e08d/examples/llava/clip.cpp#L3139-L3244

Reply:
Yeah, the logic looks fine to me.